blockly-core/core/ui/block_space/flyout.js

```diff
@@ -418,6 +418,14 @@ Blockly.Flyout.prototype.show = function(xmlList) {
   var firstBlock = xmlList && xmlList[0];
   if (firstBlock === Blockly.Variables.NAME_TYPE) {
     // Special category for variables.
+    // Allow for a mix of static + dynamic blocks. Static blocks will appear
+    // first in the category
+    for (var i = 1, xml; xml = xmlList[i]; i++) {
+      if (xml.tagName && xml.tagName.toUpperCase() === 'BLOCK') {
+        blocks.push(Blockly.Xml.domToBlock(this.blockSpace_, xml));
+        gaps.push(margin * 3);
+      }
+    }
```

Comment:
This isn't a core part of this change, but the ed team wanted to be able to prepopulate the Variables category with a set of variables (that aren't necessarily in use in the workspace). We'll now make this category be the set of provided static blocks, plus the autopopulated blocks (based off of used variables in the workspace).